### ISSUE TITLE
No class def found on API level 24

### ISSUE THREAD

**Issue by koral--** [ZacSweers/CatchUp#223]
App compiled from the current master. The only change is an AGP update to 4.1.0-alpha05 (the current canary version).

On devices with API 30 and 21 app works fine but crashes just after start on API 24.
Looks like issue with core library desugaring perhaps?

Here is a LogCat from AVD with API 24:
```
2020-04-12 15:53:42.889 2898-2898/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: io.sweers.catchup.debug, PID: 2898
    java.lang.NoClassDefFoundError: Failed resolution of: Ljava/lang/invoke/MethodHandles$Lookup;
        at retrofit2.Platform.<init>(Platform.java:62)
        at retrofit2.Platform$Android.<init>(Platform.java:115)
        at retrofit2.Platform.findPlatform(Platform.java:44)
        at retrofit2.Platform.<clinit>(Platform.java:34)
        at retrofit2.Platform.get(Platform.java:37)
        at retrofit2.Retrofit$Builder.<init>(Retrofit.java:430)
        at io.sweers.catchup.ui.bugreport.BugReportModule.provideImgurService$catchup_debug(BugReportApi.kt:79)
        at io.sweers.catchup.ui.bugreport.BugReportModule_ProvideImgurService$catchup_debugFactory.provideImgurService$catchup_debug(BugReportModule_ProvideImgurService$catchup_debugFactory.java:55)
        at io.sweers.catchup.ui.bugreport.BugReportModule_ProvideImgurService$catchup_debugFactory.get(BugReportModule_ProvideImgurService$catchup_debugFactory.java:43)
        at io.sweers.catchup.ui.bugreport.BugReportModule_ProvideImgurService$catchup_debugFactory.get(BugReportModule_ProvideImgurService$catchup_debugFactory.java:14)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at io.sweers.catchup.ui.bugreport.BugReportLens_Factory.get(BugReportLens_Factory.java:41)
        at io.sweers.catchup.ui.bugreport.BugReportLens_Factory.get(BugReportLens_Factory.java:10)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at io.sweers.catchup.ui.DebugViewContainer_Factory.get(DebugViewContainer_Factory.java:53)
        at io.sweers.catchup.ui.DebugViewContainer_Factory.get(DebugViewContainer_Factory.java:15)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at io.sweers.catchup.ui.activity.UiModule_ProvideViewContainer$catchup_debugFactory.get(UiModule_ProvideViewContainer$catchup_debugFactory.java:28)
        at io.sweers.catchup.ui.activity.UiModule_ProvideViewContainer$catchup_debugFactory.get(UiModule_ProvideViewContainer$catchup_debugFactory.java:10)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at io.sweers.catchup.app.DaggerApplicationComponent$MainActivitySubcomponentImpl.injectMainActivity(DaggerApplicationComponent.java:779)
        at io.sweers.catchup.app.DaggerApplicationComponent$MainActivitySubcomponentImpl.inject(DaggerApplicationComponent.java:774)
        at io.sweers.catchup.app.DaggerApplicationComponent$MainActivitySubcomponentImpl.inject(DaggerApplicationComponent.java:525)
        at dagger.android.DispatchingAndroidInjector.maybeInject(DispatchingAndroidInjector.java:113)
        at dagger.android.DispatchingAndroidInjector.inject(DispatchingAndroidInjector.java:134)
        at dagger.android.AndroidInjection.inject(AndroidInjection.java:181)
        at dagger.android.AndroidInjection.inject(AndroidInjection.java:55)
        at io.sweers.catchup.base.ui.InjectableBaseActivity.onCreate(InjectableBaseActivity.kt:40)
        at io.sweers.catchup.ui.activity.MainActivity.onCreate(MainActivity.kt:77)
        at android.app.Activity.performCreate(Activity.java:6662)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2599)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6077)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "java.lang.invoke.MethodHandles$Lookup" on path: DexPathList[[zip file "/data/app/io.sweers.catchup.debug-1/base.apk"],nativeLibraryDirectories=[/data/app/io.sweers.catchup.debug-1/lib/x86, /data/app/io.sweers.catchup.debug-1/base.apk!/lib/x86, /system/lib, /vendor/lib]]
2020-04-12 15:53:42.889 2898-2898/? E/AndroidRuntime:     at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        	... 41 more
```

**Comment by ZacSweers** [comment]
Thanks, I think this was fixed in a retrofit update already

**Comment by ZacSweers** [comment]
Yeah confirmed it's fixed in 2.8.1. I'll update
